ludwig/schema/prompt.py

```diff
@@ -0,0 +1,79 @@
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.utils import ludwig_dataclass
+
+
+@DeveloperAPI
+@ludwig_dataclass
+class RetrievalConfig(schema_utils.BaseMarshmallowConfig):
+    """This Dataclass is a schema for the nested retrieval config under prompt."""
+
+    type: str = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description=(
+            "The type of retrieval to use for the prompt. If `None`, then no retrieval is used, and the task "
+            "is framed as a zero-shot learning problem. If not `None` (e.g. either 'random' or 'semantic'), then "
+            "samples are retrieved from an index of the training set and used to augment the input to the model "
+            "in a few-shot learning setting."
+        ),
+    )
+
+    index_name: str = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="The name of the index to use for the prompt. Indices are stored in the ludwig cache by default.",
+    )
+
+    model_name: str = schema_utils.String(
+        default=None,
```

Comment:
If we're going to leave this as None, then we should make sure there's some sort of validation check somewhere to raise an error at the time of building the retrieval index with a useful error message.

In hindsight, I should probably add that for LLM creation as well to `model_name` make sure a value is passed in

Reply:
Added a config validation check for this